numpy/core/src/multiarray/scalartypes.c.src

```diff
@@ -1549,6 +1549,16 @@ gentype_@name@(PyObject *self, PyObject *args, PyObject *kwds)
 }
 /**end repeat**/
 
+static PyObject *
+dunder_round(PyObject *self, PyObject *args, PyObject *kwds)
+{
+    PyObject *obj = gentype_round(self, args, kwds);
```

Comment:
Does this upcast int to float,  losing precision?

Reply:
No, it's equivalent to `scalar.round`.

```python
>>> type(np.int64(5).round())
<class 'numpy.int64'>
```